ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,1027 @@
+use bitflags::bitflags;
+use ctru_sys::Handle;
+
+pub struct Cam(());
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamPort: u32 {
+        const PORT_NONE = ctru_sys::PORT_NONE;
+        const PORT_CAM1 = ctru_sys::PORT_CAM1;
+        const PORT_CAM2 = ctru_sys::PORT_CAM2;
+        const PORT_BOTH = ctru_sys::PORT_BOTH;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamSelect: u32 {
+        const SELECT_NONE      = ctru_sys::SELECT_NONE;
+        const SELECT_OUT1      = ctru_sys::SELECT_OUT1;
+        const SELECT_IN1       = ctru_sys::SELECT_IN1;
+        const SELECT_OUT2      = ctru_sys::SELECT_OUT2;
+        const SELECT_IN1_OUT1  = ctru_sys::SELECT_IN1_OUT1;
+        const SELECT_OUT1_OUT2 = ctru_sys::SELECT_OUT1_OUT2;
+        const SELECT_IN1_OUT2  = ctru_sys::SELECT_IN1_OUT2;
+        const SELECT_ALL       = ctru_sys::SELECT_ALL;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamContext: u32 {
+        const CONTEXT_NONE = ctru_sys::CONTEXT_NONE;
+        const CONTEXT_A    = ctru_sys::CONTEXT_A;
+        const CONTEXT_B    = ctru_sys::CONTEXT_B;
+        const CONTEXT_BOTH = ctru_sys::CONTEXT_BOTH;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamFlip: u32 {
+        const FLIP_NONE       = ctru_sys::FLIP_NONE;
+        const FLIP_HORIZONTAL = ctru_sys::FLIP_HORIZONTAL;
+        const FLIP_VERTICAL   = ctru_sys::FLIP_VERTICAL;
+        const FLIP_REVERSE    = ctru_sys::FLIP_REVERSE;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamSize: u32 {
+        const SIZE_VGA            = ctru_sys::SIZE_VGA;
+        const SIZE_QVGA           = ctru_sys::SIZE_QVGA;
+        const SIZE_QQVGA          = ctru_sys::SIZE_QQVGA;
+        const SIZE_CIF            = ctru_sys::SIZE_CIF;
+        const SIZE_QCIF           = ctru_sys::SIZE_QCIF;
+        const SIZE_DS_LCD         = ctru_sys::SIZE_DS_LCD;
+        const SIZE_DS_LCD_X4      = ctru_sys::SIZE_DS_LCDx4;
+        const SIZE_CTR_TOP_LCD    = ctru_sys::SIZE_CTR_TOP_LCD;
+        const SIZE_CTR_BOTTOM_LCD = ctru_sys::SIZE_CTR_BOTTOM_LCD;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamFrameRate: u32 {
+        const FRAME_RATE_15       = ctru_sys::FRAME_RATE_15;
```

Comment:
Since the enum is named CamFrameRate, you can remove the `FRAME_RATE_` prefix from all variants.